app/src/main/java/com/example/photofy/DetectProperties.java

```diff
@@ -29,77 +31,76 @@ public class DetectProperties {
 
     public static final String TAG = "DetectProperties";
 
-    private Photo picture;
-    private String path;
     private GoogleCredentials credentials;
-    private List<String> objects;
-
-    public DetectProperties(Photo picture, String path) {
-        this.picture = picture;
-        this.path = path;
-        objects = new ArrayList<String>();
+    private ImageAnnotatorClient client;
+
+    public DetectProperties(String credentialsFilePath, Context context) {
+        try {
+            // Authorize Google credentials
+            credentials = GoogleCredentials.fromStream(context.getAssets().open(credentialsFilePath))
+                    .createScoped(Arrays.asList("https://www.googleapis.com/auth/cloud-platform"));
+
+            // Create annotator client
+            ImageAnnotatorSettings imageAnnotatorSettings =
+                    ImageAnnotatorSettings.newBuilder()
+                            .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+                            .build();
+            client = ImageAnnotatorClient.create(imageAnnotatorSettings);
+        } catch (IOException e) {
+            Log.e(TAG, "credentials error" + e);
+        }
     }
 
-    public String authExplicit(String jsonPath, Context context) throws IOException {
-        // Authorize Google credentials
-        credentials = GoogleCredentials.fromStream(context.getAssets().open(jsonPath))
-                .createScoped(Arrays.asList("https://www.googleapis.com/auth/cloud-platform"));
-        Storage storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
-
+    public void findDominantColor(Photo picture, String path) throws IOException {
         // Upload captured image to Google Cloud storage
+        Storage storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
         String bucketName = "photofy-images0";
         String objectName = "image-" + picture.getImage().getName();
         UploadObject.uploadObject(storage, bucketName, objectName, path);
 
-        // Get GCS path for image from bucket
-        return "gs://" + bucketName + "/" + objectName;
+        // GCS path for image from bucket
+        String gcsPath = "gs://" + bucketName + "/" + objectName;
+
+        // Save generated dominant color
+        String hex = detectPropertiesGcs(gcsPath);
+        picture.setColor(hex);
```

Comment:
Return the color from this method and let the caller use the color object(string)in whichever way it wants to use. Try not to modify the input parameters of a function and return result of the function as output.